lib/Transformation.ts

```diff
@@ -76,6 +76,11 @@ function transformWildcard(term: Alg.WildcardExpression): E.Expression {
 }
 
 // TODO: Maybe do this with a map?
+/**
+ * Some of the types here have some super relation.
+ * This should match the relations provided in @see{extensionTableInput}.
+ * @param lit
+ */
 export function transformLiteral(lit: RDF.Literal): E.Literal<any> {
   if (!lit.datatype) {
     return lit.language ?
```

Comment:
I was wondering if we could get some kind of reference on why we do this? It actually reminds me a bit of #14 ? Anyway, it's peculiar and I think we should write down what happens here.

Reply:
Same thing for the cases of null, undefined and '' underneath this.

Reply:
You mean the fact that if there's no datatype this is considered a string?

Reply:
Yes.
Also in https://www.w3.org/TR/sparql11-query/#func-lang I found
> Note that the RDF data model does not include literals with an empty language tag.

Does this have anything to do with this?

Reply:
It's this one again: https://www.w3.org/TR/sparql11-query/#func-string.

Reply:
Or at least largely. I don't think there are any functions that take an xsd:string but not a simple literal (no datatype).

Reply:
I think there has been some discussion in IDLab regarding this as well, and possible spec incompatibilities. Maybe @rubensworks has some better references.

Reply:
See also https://www.w3.org/TR/2004/REC-rdf-concepts-20040210/#dfn-plain-literal .

The distinction between simple literal and plain literal has never been clear to be, as plain literals are simple literals but with an (optional) language tag. So a plain one without a language tag is just a simple one?

The key part is simple, explicitly typed, and language fall under "string literal" (i.e. stringly in our case), and there are IIRC functions that take an explicitly typed xsd:string but not a simple (no datatype, no language) one.

There are functions that take xsd:string + simple, but don't take languageStrings.

Reply:
This issue should contain all the answers :-) https://github.com/w3c/sparql-12/issues/112

Reply:
Ah exactly. And for reference, we already do RDF 1.1 behavior. If there is no datatype, we make it xsd:string. (except when there is a language tag ofc)

Reply:
Alright, thank you for this reference. Now I will add this reference to the code. Should I add a counter as well? :p (jk)